build-logic-commons/gradle-plugin/src/main/kotlin/CacheMissMonitorBuildService.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.SetProperty
+import org.gradle.api.services.BuildService
+import org.gradle.api.services.BuildServiceParameters
+import org.gradle.tooling.events.FinishEvent
+import org.gradle.tooling.events.OperationCompletionListener
+import org.gradle.tooling.events.task.TaskFinishEvent
+import org.gradle.tooling.events.task.TaskSuccessResult
+import javax.inject.Inject
+
+abstract class CacheMissMonitorBuildService @Inject constructor(objects: ObjectFactory) : BuildService<CacheMissMonitorBuildService.Params>, OperationCompletionListener {
+    val cacheMiss: Property<Boolean> = objects.property(Boolean::class.java).convention(false)
```

Comment:
I am not sure it makes sense to use a property here. This is only used during execution to capture whether there was a cache miss or not. Maybe an `AtomicBoolean` or simple a volatile `boolean` would be better.

Reply:
Yeah.. that's what comes into my mind at first, too...

Reply:
Since you use it for wiring, I think a `Property` is better after all. I thought we could go away without wiring those things together though we can't.